atd-vze/src/views/Crashes/CrashChangeLog.js

```diff
@@ -1,243 +1,177 @@
-      // human-readble output.
-      if (typeof item.original_record_value === "object") {
-        item.original_record_value = JSON.stringify(
-          item.original_record_value,
-          stringifyReplacer
-        );
-      }
-
-      if (typeof item.archived_record_value === "object") {
-        item.archived_record_value = JSON.stringify(
-          item.archived_record_value,
-          stringifyReplacer
-        );
-      }
-
-      return (
-        <tr key={`recordHistory-${i}`} className="d-flex">
-          <td className="col-2 text-break">{item.original_record_key}</td>
-          <td className="col-5">
-            <Badge color="primary" className="text-wrap text-break">
-              {String(item.original_record_value)}
-            </Badge>
-          </td>
-          <td className="col-5">
-            <Badge color="danger" className="text-wrap text-break">
-              {String(item.archived_record_value)}
-            </Badge>
-          </td>
-        </tr>
+    return diffs;
+  }, []);
+  // sort array of entries by field name
+  diffArray.sort((a, b) => {
+    if (a.field < b.field) {
+      return -1;
+    }
+    if (a.field > b.field) {
+      return 1;
+    }
+    return 0;
+  });
+  return diffArray;
+};
+
+/**
+ * Hook that identifies that returns an array with one entry per row in the
+ * the change log view for the given crash. Each object in the returned
+ * array has two properties:
+ * - diffs: an array of old/new values for each field that has changed
+ * - affected_fields: an array of the field names that have changed
+ */
+const useChangeLogData = data =>
+  useMemo(() => {
+    if (!data) {
+      return [];
+    }
+    return data.map(change => {
+      change.diffs = getDiffArray(
+        change.record_json.old,
+        change.record_json.new
       );
+      change.affected_fields = change.diffs.map(diff => diff.field);
+      return change;
     });
-
-    // Generate the body of the modal box
-    modalBody = (
-      <section>
-        <h6>Crash ID: {record.record_crash_id}</h6>
-        <h6>Edited Date: {formatDateTimeString(record.update_timestamp)}</h6>
-        <h6>Updated by: {record.updated_by || "Unavailable"}</h6>
-        &nbsp;
-        <Table responsive className="overflow-hidden">
+  }, [data]);
+
+const isNewRecordEvent = change => change.operation_type === "create";
+
+/**
+ * Header component of the change details table
+ */
+const ChangeSummary = ({ selectedChange }) => {
+  return (
+    <>
+      {`${selectedChange.record_type}`} ID{" "}
+      <span className="font-monospace">{selectedChange.record_id}</span> edited
+      by {selectedChange.created_by}
+      {" - "}
+      {formatDateTimeString(selectedChange.created_at)}
+    </>
+  );
+};
+
+/**
+ * Modal which renders change details when a change log entry is clicked
+ */
+const ChangeDetailsModal = ({ selectedChange, setSelectedChange }) => {
```

Comment:
would y'all prefer this in a separate file?

Reply:
hmm yeah i think it would make sense organizationally for it to be a diff file

Reply:
done! 🙏